core/src/test/scala/spinal/core/PhysicalNumbers.scala

```diff
@@ -0,0 +1,37 @@
+package spinal.core
+
+import org.scalatest.funsuite.AnyFunSuite
+
+class PhysicalNumbers extends AnyFunSuite {
+  val hz100 = 100 Hz
+  val hz100_2 = 100 Hz
+  val hz200 = 200 Hz
+  val ms2 = 100 sec
+
+  test("Comparisons") {
+    assert(hz100 == hz100)
+    assert(hz100 == hz100_2)
+    assert(hz100_2 == hz100)
+    assert(hz100 != hz200)
+    assert(hz200 != ms2)
+    assert(hz100 != null)
+
+    assert(hz100 < hz200)
+    assert(!(hz200 < hz100))
+    assert(!(hz100 < hz100_2))
+
+    assert(hz200 > hz100)
+    assert(!(hz100 > hz200))
+    assert(!(hz100 < hz100_2))
+
+    assert(hz100 <= hz200)
+    assert(hz100 <= hz100_2)
+    assert(!(hz200 <= hz100))
+
+    assert(hz100 >= hz100_2)
+    assert(hz200 >= hz100)
+    assert(!(hz100 >= hz200))
+
+    assert(hz100.hashCode() == hz100_2.hashCode())
```

Comment:
```suggestion
    assert(hz100 == hz100_2)
```

It feels more natural

Reply:
That is the same as the check further above, here I wanted to explicitly check that hashCode is implemented 'correctly' (for some approximation of correctness ;-))

Reply:
Ah sorry, didn't see